src/frontend/src/scheduler/distributed/stage.rs

```diff
@@ -289,14 +292,44 @@ impl StageRunner {
 
     async fn schedule_tasks(&self) -> SchedulerResult<()> {
         let mut futures = vec![];
-        for id in 0..self.stage.parallelism {
-            let task_id = TaskIdProst {
-                query_id: self.stage.query_id.id.clone(),
-                stage_id: self.stage.id,
-                task_id: id,
-            };
-            let plan_fragment = self.create_plan_fragment(id);
-            futures.push(async { self.schedule_task(task_id, plan_fragment).await });
+
+        if let Some(table_scan_info) = self.stage.table_scan_info.as_ref() {
```

Comment:
What about add some comments here why we need a if-else branch here?

Reply:
added